### PR TITLE
[VIRTS-2004] Plugins API v2

### PR DESCRIPTION
**Pull request by bleepbop**
## Description

Created new API endpoint for retrieving plugins.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

I made GET requests to the following endpoints as both a red and a blue user:
* /api/v2/plugins
* /api/v2/plugins/emu (can test with any plugin name)

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: app/api/v2/handlers/plugins_api.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @aiohttp_apispec.querystring_schema(BaseGetOneQuerySchema)
+    @aiohttp_apispec.response_schema(PluginSchema)
+    async def get_plugin_by_name(self, request: web.Request):
+        plugin_name = request.match_info['plugin_name']
+        include = request['querystring'].get('include')
+        exclude = request['querystring'].get('exclude')
+
+        search = dict(name=plugin_name)
+
+        plugin = self._api_manager.get_object_with_filters('plugins', search=search, include=include, exclude=exclude)
```
Comment (b17zr) [inline app/api/v2/handlers/plugins_api.py:42]: Does searching with access yield different results? If this was copied from planners -- they are a special case. See https://github.com/mitre/caldera/pull/2167/files#diff-8891c148b883a0802ff3991493a07472f78369ab22ecd01c8bfc2816acc59332R43 for an example.

**Review by b17zr — CHANGES_REQUESTED** [review]
Looks great! One comment on searching with plugin access.

File: app/api/v2/handlers/plugins_api.py (hunk 1 above)
Comment (bleepbop) [reply app/api/v2/handlers/plugins_api.py]: Thanks! I made the change, now only the plugins associated with the Red/Blue status of the user should be returned

**Review by b17zr — APPROVED** [review]
LGTM!

**Review by wbooth — DISMISSED** [review]

**Review by uruwhy — APPROVED** [review]
Looks good!

**Review by wbooth — APPROVED** [review]

**Pull request merged**